agent/config/config.go

```diff
@@ -238,7 +239,8 @@ func environmentConfig() Config {
 	// invalid parse
 	// Blank is not a warning; we have sane defaults
 	if err != io.EOF && err != nil {
-		seelog.Warnf("Invalid format for \"ECS_RESERVED_PORTS\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		err := fmt.Errorf("Invalid format for \"ECS_RESERVED_PORTS\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		seelog.Warn(err)
```

Comment:
We've decided to leave these as warnings, at least for now, so they shouldn't be appended to `errs`, only logged.